api/envoy/extensions/filters/http/router/v3/router.proto

```diff
@@ -134,4 +134,10 @@ message Router {
   // upstream HTTP filters will count as a final response if hedging is configured.
   // [#extension-category: envoy.filters.http.upstream]
   repeated network.http_connection_manager.v3.HttpFilter upstream_http_filters = 8;
+
+  // If set to true, Envoy will reject ``CONNECT`` requests that send data before
+  // receiving a ``200`` response from the upstream. This early data behavior
+  // is common for latency reduction but can cause issues with some upstreams.
+  // Defaults to false to allow early data and be compatible with common behavior.
+  bool reject_connect_request_early_data = 10;
```

Comment:
Suggest using `google.protobuf.BoolValue` here, so that different data planes could have different defaults.

Reply:
Done.